### PR TITLE
Add details supporting existing resources

### LINKED ISSUE #1577: [CSS]: General request - Improve documentation so that it helps to answer the question "I already have resources that I want to bring to Aspire, and don't want Aspire to be creating them for me"
### Article title

N/A

### Article URL

N/A

### Description

As I go over the documentation, in many places it is hard to understand / not clear or simple enough how can the developer that is just trying out Aspire bring their own resources into Aspire rather than to have Aspire create them on the developer's behalf.

This is a general perception I have towards the existing subset of documentation, mainly when it comes to database creation and other resources such as Key Vault.

---
[Associated WorkItem - 308086](https://dev.azure.com/msft-skilling/Content/_workitems/edit/308086)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

- Add links to tutorial on existing projects.
- Add details supporting existing resources.

Fixes #1577
/cc @HugoDurana

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/fundamentals/app-host-overview.md](https://github.com/dotnet/docs-aspire/blob/342eec1fa2688a8af96a521d80eb0f8e3bbcdeb5/docs/fundamentals/app-host-overview.md) | [docs/fundamentals/app-host-overview](https://review.learn.microsoft.com/en-us/dotnet/aspire/fundamentals/app-host-overview?branch=pr-en-us-1691) |

### CODE REVIEW

File: docs/fundamentals/app-host-overview.md [hunk 1]
```diff
@@ -1,31 +1,42 @@
 ---
 title: .NET Aspire orchestration overview
-description: Learn the fundamental concepts of .NET Aspire orchestration and explore the various APIs to express resource references.
-ms.date: 09/24/2024
+description: Learn the fundamental concepts of .NET Aspire orchestration and explore the various APIs for expressing and referencing resources.
```
Comment (tdykstra) [inline docs/fundamentals/app-host-overview.md:3]: Not clear to me what "expressing" means in this context.

File: docs/fundamentals/app-host-overview.md [hunk 2]
```diff
@@ -1,31 +1,42 @@
...
+.NET Aspire empowers you to seamlessly build, provision, deploy, configure, test, run, and observe your distributed applications. All of these capabilities are achieved through the utilization of an _app model_ that outlines the resources in your .NET Aspire solution and their relationships. These resources encompass projects, executables, containers, and external services and cloud resources that your app depends on. Within every .NET Aspire solution, there's a designated [App host project](#app-host-project), where the app model is precisely defined using methods available on the <xref:Aspire.Hosting.IDistributedApplicationBuilder>. This builder is obtained by invoking <xref:Aspire.Hosting.DistributedApplication.CreateBuilder%2A?displayProperty=nameWithType>.
+
+```csharp
+// Create a new app model builder
+var builder = DistributedApplication.CreateBuilder(args);
+
+// TODO:
```
Comment (tdykstra) [inline docs/fundamentals/app-host-overview.md:34]: Did you intend to leave this TODO here?

**Review by tdykstra — APPROVED** [review]
LGTM, just a couple questions.

File: docs/fundamentals/app-host-overview.md (hunk 2 above)
Comment (IEvangelist) [reply docs/fundamentals/app-host-overview.md:34]: Yes, it's for the reader....

File: docs/fundamentals/app-host-overview.md (hunk 1 above)
Comment (IEvangelist) [inline docs/fundamentals/app-host-overview.md:3]:
```suggestion
description: Learn the fundamental concepts of .NET Aspire orchestration and explore the various APIs for adding resources and expressing dependencies.
```

**Pull request merged**